packages/salesforcedx-vscode-lwc/resources/static/js-meta.xsd

```diff
@@ -0,0 +1,291 @@
+                            </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="lightningSnapin__PreChat">
+              <xs:annotation>
+                <xs:documentation>
+                                Enables a custom prechat component to be selected from Embedded Service Chat Setup. A component that imports the lightningsnapin/basePrechat module must specify the lightningSnapin__PreChat target. For more information, see the lightningsnapin/basePrechat documentation.
+                            </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="lightningSnapin__ChatHeader">
+              <xs:annotation>
+                <xs:documentation>
+                                Enables a custom chat header component to be selected from Embedded Service Chat Setup. A component that imports the lightningsnapin/baseChatHeader module must specify that lightningSnapin__ChatHeader target. For more information, see the lightningsnapin/baseChatHeader documentation.
+                            </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+
+  <!-- objects SXD-->
+  <xs:complexType name="objects">
+    <xs:annotation>
+      <xs:documentation>
+        Limits the component to a set of one or more objects. If you don’t use this tag set, the component supports all supported objects. This tag set works only inside a parent targetConfig that’s configured for lightning__RecordPage.Specify the objects tag set only once inside a targetConfig set. Supports the object subtag.
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:element type="xs:string" name="object" maxOccurs="unbounded" minOccurs="0">
+        <xs:annotation>
+          <xs:documentation>
+            Defines which objects the component is supported for. Use one object tag for each supported object. See the User Interface API Developer Guide for the list of supported objects.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <!-- property SXD-->
+  <xs:complexType name="property">
+    <xs:annotation>
+      <xs:documentation>
+                Specifies a public property of a component that can be set in Lightning App Builder, App Manager, Lightning Flow Builder, or Experience Builder. The component author defines the property in the component’s JavaScript class using the @api decorator.
+          </xs:documentation>
+    </xs:annotation>
+    <xs:attribute type="xs:string" name="datasource" use="optional" />
```

Comment:
Is there any way to add doc for xs:attribute tags? Why use xs:attribute here instead of xs:element used in other complexType definitions?

Reply:
Yes, I think there is a way, let me update that.
The reason is attributes lives inside the xml tag while elements are children of the tag.